### PR TITLE
fix map stop loading error

### PR DESCRIPTION
**Pull request by Huibean**
fix tile files download with ssl error in python3

### CODE REVIEW

**Comment by peterbarker** [comment]
Couldn't we import this as `url_error_method` then define `url_method` in terms of that?

**Comment by Huibean** [comment]
> 
Hi @peterbarker ,you mean separated error exception and use same method(change tile services url to next), but the problem is bot the exception catch code run under python2 and python3, well python2 haven't define this exception

**Comment by Huibean** [comment]
Replenish the original error, map stop loading while unexpected error raised
```shell
MAV> Exception in thread Thread-1:
Traceback (most recent call last):
  File "/usr/lib/python3.7/threading.py", line 917, in _bootstrap_inner
    self.run()
  File "/usr/lib/python3.7/threading.py", line 865, in run
    self._target(*self._args, **self._kwargs)
  File "/home/parallels/.local/lib/python3.7/site-packages/MAVProxy/modules/mavproxy_map/mp_tile.py", line 288, in downloader
    resp = url_open(req)
  File "/usr/lib/python3.7/urllib/request.py", line 222, in urlopen
    return opener.open(url, data, timeout)
  File "/usr/lib/python3.7/urllib/request.py", line 525, in open
    response = self._open(req, data)
  File "/usr/lib/python3.7/urllib/request.py", line 543, in _open
    '_open', req)
  File "/usr/lib/python3.7/urllib/request.py", line 503, in _call_chain
    result = func(*args)
  File "/usr/lib/python3.7/urllib/request.py", line 1352, in http_open
    return self.do_open(http.client.HTTPConnection, req)
  File "/usr/lib/python3.7/urllib/request.py", line 1327, in do_open
    r = h.getresponse()
  File "/usr/lib/python3.7/http/client.py", line 1352, in getresponse
    response.begin()
  File "/usr/lib/python3.7/http/client.py", line 310, in begin
    version, status, reason = self._read_status()
  File "/usr/lib/python3.7/http/client.py", line 279, in _read_status
    raise RemoteDisconnected("Remote end closed connection without"
http.client.RemoteDisconnected: Remote end closed connection without response
```

File: MAVProxy/modules/mavproxy_map/mp_tile.py [hunk 1]
```diff
@@ -35,10 +35,13 @@
     from urllib2 import Request as url_request
     from urllib2 import urlopen as url_open
     from urllib2 import URLError as url_error
+    download_error = url_error
 else:
     from urllib.request import Request as url_request
     from urllib.request import urlopen as url_open
     from urllib.error import URLError as url_error
+    from http.client import RemoteDisconnected
+    download_error = (RemoteDisconnected, url_error)
```
Comment (peterbarker) [inline MAVProxy/modules/mavproxy_map/mp_tile.py:44]:
```suggestion
    from urllib.error import URLError as actual_url_error
    from http.client import RemoteDisconnected
    url_error = (RemoteDisconnected, actual_url_error)
```

File: MAVProxy/modules/mavproxy_map/mp_tile.py [hunk 2]
```diff
@@ -35,10 +35,13 @@
     from urllib2 import Request as url_request
     from urllib2 import urlopen as url_open
     from urllib2 import URLError as url_error
+    download_error = url_error
```
Comment (peterbarker) [inline MAVProxy/modules/mavproxy_map/mp_tile.py:38]:
```suggestion
```

File: MAVProxy/modules/mavproxy_map/mp_tile.py [hunk 3]
```diff
@@ -287,7 +290,7 @@ def downloader(self):
                     req.add_header('Referer', 'https://maps.google.com/')
                 resp = url_open(req)
                 headers = resp.info()
-            except url_error as e:
+            except download_error as e:
```
Comment (peterbarker) [inline MAVProxy/modules/mavproxy_map/mp_tile.py:293]:
```suggestion
            except url_error as e:
```

**Review by peterbarker — COMMENTED** [review]
This is what I meant...

File: MAVProxy/modules/mavproxy_map/mp_tile.py (hunk 1 above)
Comment (Huibean) [reply MAVProxy/modules/mavproxy_map/mp_tile.py:44]: Get it , let me change it with better name

**Comment by Huibean** [comment]
@peterbarker OK, commit updated

**Comment by peterbarker** [comment]
@Huibean tested under both Python3 I assume?  No need to test Python2 now - this now can't break it, which is handy.

**Comment by Huibean** [comment]
> @Huibean tested under both Python3 I assume? No need to test Python2 now - this now can't break it, which is handy.

yes, according to python documnet **RemoteDisconnected** started from first python3 version

**Comment by peterbarker** [comment]
> > @Huibean tested under both Python3 I assume? No need to test Python2 now - this now can't break it, which is handy.
> 
> yes, according to python documnet **RemoteDisconnected** started from first python3 version

Sure - but have you made sure that this new code works?

**Comment by Huibean** [comment]
> 

@peterbarker yes I have run this code and works good, by the way, is there any wiki of this repo can show how to run this code from repo? I have to modify the installed code file in system to debug and install it after repo code changed to confirm
```shell
➜  MAVProxy git:(fix-tile-files-download-error) ipython MAVProxy/mavproxy.py
---------------------------------------------------------------------------
ImportError                               Traceback (most recent call last)
~/Work/MAVProxy/MAVProxy/mavproxy.py in <module>
     35 from builtins import input
     36
---> 37 from MAVProxy.modules.lib import textconsole
     38 from MAVProxy.modules.lib import mp_util
     39 from MAVProxy.modules.lib import rline

ImportError: bad magic number in 'MAVProxy': b'\x03\xf3\r\n'
```

**Comment by peterbarker** [comment]
> > 
> 
> @peterbarker yes I have run this code and works good, by the way, is there any wiki of this repo can show how to run this code from repo? I have to modify the installed code file in system to debug and install it after repo code changed to confirm
> 
> ```shell
> ➜  MAVProxy git:(fix-tile-files-download-error) ipython MAVProxy/mavproxy.py
> ---------------------------------------------------------------------------
> ImportError                               Traceback (most recent call last)
> ~/Work/MAVProxy/MAVProxy/mavproxy.py in <module>
>      35 from builtins import input
>      36
> ---> 37 from MAVProxy.modules.lib import textconsole
>      38 from MAVProxy.modules.lib import mp_util
>      39 from MAVProxy.modules.lib import rline
> 
> ImportError: bad magic number in 'MAVProxy': b'\x03\xf3\r\n'
> ```

I just always install from source with `python setup.py build install --user --force`

**Comment by peterbarker** [comment]
Merged, thanks!

**Pull request merged**